gen/nested.cpp

```
@@ -307,7 +307,8 @@ LLValue *DtoNestedContext(const Loc &loc, Dsymbol *sym) {
     IF_LOG Logger::println("Current function is %s", ctxfd->toChars());
     if (fromParent) {
       ctxfd = getParentFunc(ctxfd);
-      assert(ctxfd && "Context from outer function, but no outer function?");
+      if (!ctxfd)
+        ctxfd = irFunc.decl;
```

add log output about this strange case?

---

does this actually produce correct code now?

---

It doesn't crash, which is what it did before. As for the correctness of the code,I'm not really sure, but I don't know what its supposed to do other than iterate through the linked list of classes. @ZILtoid1991 could you provide a reduced test case of this code in action?

---

```d
class DOMImplementation {
class Attr  {  Attr  _nextAttr; }
class Map  {
    Attr firstAttr;
    auto opSlice()
    {
         struct Range
         {
             Attr currentAttr;

             auto front() { return currentAttr; }
             void popFront() { currentAttr = currentAttr._nextAttr; }
             bool empty() { return currentAttr is null; }
        }
       return Range(firstAttr);
   }
}
}
```

---

Thx, somewhat further reduced:
```D
class Outer {
    class Inner {
        auto opSlice() {
            struct Range {
                void foo() {}
            }
            return Range();
        }
    }
}
```

**2** outer aggregates are apparently required, and both need to be classes (works with any one of them being a struct).

---

> It doesn't crash, which is what it did before. As for the correctness of the code,I'm not really sure, but I don't know what its supposed to do other than iterate through the linked list of classes. 

If unsure about the correct behavior, then better to exit gracefully instead of silently continuing.